### ISSUE TITLE
Xcode 11 support ?

### ISSUE THREAD

**Issue by jhjulkun** [laanlabs/HomeRobot#1]
Hi
So after spending some time looking over the WWDC videos for ARKit I then googled ARKit for robotics and this was the top result! 

Seems like a very promising start!
However, Xcode 11 now uses the Swift5 compiler, and apparently the PlaceNoteSDK is in Swift4 

As a workaround you can install Xcode 9.4.1 and the repository compiles/runs, but since I don't have the Romotive robot I'm working on somehow using the the same RMCore but with my robotics.

Any case, I'll keep chipping away and see if I can get something working, but if you guys have some time to test with the latest Xcode that would be great.
Thanks

**Comment by ferdinandl007** [comment]
@jhjulkun did You make any progress on getting this to run on xcode 11?

**Comment by jhjulkun** [comment]
No, the Placenote SDK for iOS has build issues for Xcode11/iOS13, but the Placenote team is working to resolve it:
https://github.com/Placenote/PlacenoteSDK-iOS/issues/66

However, if you just want to use an iOS13 device, install the device support files for it:
https://stackoverflow.com/questions/51215836/ios-12-not-supported-by-xcode-9-4-could-not-locate-device-support-files
https://github.com/filsv/iPhoneOSDeviceSupport

**Comment by ferdinandl007** [comment]
@jhjulkun Thanks for the response 
 I looked through the code  and  that particular API only seems to be used for re-localisation and  map sharing it should be possible to use the native re-localisation with a ARKit supports since IOS 12  and for the map sharing it should be possible to use iCloud or Google  fire base.  I will probably work on we factoring this let me know if you want to help out and this to :)

**Comment by jhjulkun** [comment]
An all native ARKit solution would be nice! 
I have been looking at another similar solution (also using Placenote, but for Unity), and Unity is interesting but its certainly not fast to build for iOS, or maybe I just need a faster computer! :)
I'm however new to 3D in general so both ARKit and Unity has been a steep ramp.
https://www.youtube.com/watch?v=VOMysKbDNxk

**Comment by ferdinandl007** [comment]
@jhjulkun  ok Will try reconfiguring the application for peer-to-peer map sharing using a ARKit  only should be possible.  additionally I will add Voxel mapping and anonymous navigation for my bachelor's thesis so going to be a pretty fun project overall.

**Comment by jhjulkun** [comment]
Excellent! In the meantime, Placenote just released 1.8 so I'll try updating to that when I have time.
https://docs.placenote.com/whats-new-in-sdk-1.8

**Comment by cclaan** [comment]
Hey, 
Sorry about the issues on the repo -- it's not something we have looked at in a while, and things are a bit busy right now.  Thanks for debugging things; happy to accept pull requests. 
ARKit world mapping would be nice.  Are you planning to use a Romo robot? Are you able to buy them?

**Comment by jhjulkun** [comment]
Yes, you can find Romotive robots on eBay, however I'm currently using another tank type chassis for my robot: https://www.amazon.com/gp/product/B07BHFM9PJ/ref=ppx_yo_dt_b_asin_title_o09_s03?ie=UTF8&psc=1

Plus an esp32 arduino that talk BLE to iOS.

I should however be able to modify the Romo SDK to talk to my rover, but for the long run, maybe make it easier to talk to any sort of rover ? (maybe add a simple abstraction layer with easily modifiable drive commands ?)

I did order a Romotive robot, but will not be able to pick it up until Christmas.

**Comment by cclaan** [comment]
Cool, I'd be interested to see your final setup and what it costs. Seems like there should be a cheap open source solution for this since it's not terribly complicated.

**Comment by jhjulkun** [comment]
Not a bad idea. Once I get something working I'll setup something on github.

**Comment by ferdinandl007** [comment]
Yes I managed to get a hold of a Romotive robot.  pretty cool SDK
 anyway I sent you a  pull request https://github.com/laanlabs/HomeRobot/pull/2 with the new updates  now it supports iOS 13 and  native ARKit  world sharing and relocalization.
additionally will post an update on this thread once I  got the autonomous indoor navigation sorted out.
